tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,446 @@
+)
+
+from .conftest import BleBoxTestHelper, mock_feature
+
+
+class CoverTestHelper(BleBoxTestHelper):
+    """Shared test helpers for Cover tests."""
+
+    HASS_TYPE = cover
+
+
+def shutterbox_data():
+    """Return a default cover entity mock."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-shutterBox-2bee34e750b8-position",
+        full_name="shutterBox-position",
+        device_class="shutter",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My shutter")
+    type(product).model = PropertyMock(return_value="shutterBox")
+    return CoverTestHelper(feature)
+
+
+def gatebox_data():
+    """Return a default gatebox cover entity mock."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateBox-1afe34db9437-position",
+        device_class="gatebox",
+        full_name="gateBox-position",
+        current=None,
+        state=None,
+        has_stop=False,
+        is_slider=False,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gatebox")
+    type(product).model = PropertyMock(return_value="gateBox")
+    return CoverTestHelper(feature)
+
+
+def gatecontroller_data():
+    """Return a default gateController cover entity mock."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateController-2bee34e750b8-position",
+        full_name="gateController-position",
+        device_class="gate",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gate controller")
+    type(product).model = PropertyMock(return_value="gateController")
+    return CoverTestHelper(feature)
+
+
+@pytest.fixture
+def shutterbox():
+    """Return a shutterBox fixture."""
+    return shutterbox_data()
+
+
+@pytest.fixture
+def gatebox():
+    """Return a gateBox fixture."""
+    return gatebox_data()
+
+
+@pytest.fixture
+def gatecontroller():
+    """Return a gateController fixture."""
+    return gatecontroller_data()
+
+
+@pytest.fixture(params=[gatecontroller_data, gatebox_data, shutterbox_data])
+def all_cover_types(request):
+    """Return a fixture using all cover types."""
+    return request.param()
+
+
+@pytest.fixture(params=[gatecontroller_data, shutterbox_data])
+def all_sliders(request):
+    """Return a fixture using all positionable types."""
+    return request.param()
+
+
+@pytest.fixture(params=[gatebox_data])
+def not_sliders(request):
+    """Return a fixture using all non-positionable types."""
+    return request.param()
+
+
+def assert_state(entity, state):
+    """Assert that cover state is correct."""
+    assert entity.state == state
```

Comment:
Don't interact with the entity directly. Get the state of the entity from the core state machine and assert the core state. The state is the truth of the entity.

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/0519b96bec79b2db2ec8021aca1149288e2b659a/tests/components/heos/test_media_player.py#L66-L100